docs-src/http-api.md

```diff
@@ -0,0 +1,530 @@
+
+### Request
+`POST /api/v0/records`
+
+Here, `:docid` should be replaced by the string representation of the DocID of the document that is being requested.
+
+
+
+The request body should contain the following fields:
+
+- `docId` - the name of the doctype to use, string
+- `record` - the genesis content of the document (will differ per doctype)
+- `docOpts` - options for the document creation [[DocOpts]]
+
+### Response
+
+* `docId` - the [[DocID]] of the document that was modified
+* `state` - the new state of the document that was modified, [[DocState]]
+
+### Example
+
+```bash
+
+curl http://localhost:7007/api/v0/records -X POST -d '{
+  "docId": "kjzl6cwe1jw14ahmwunhk9yjwawac12tb52j1uj3b9a57eohmhycec8778p3syv",
+  "record": {
+    "jws": {
+      "payload": "AXESINm6lI30m3j5H2ausx-ulXj-L9CmFlOTZBZvJ2O734Zt",
+      "signatures": [
+        {
+          "signature": "zsLJbBSU5xZTQkYlXwEH9xj_t_8frvSFCYs0SlVMPXOnw8zOJOsKnJDQlUOvPJxjt8Bdc_7xoBdmcRG1J1tpCw",
+          "protected": "eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa2ZaNlM0TlZWVEV1dHM4bzV4RnpSTVI4ZUM2WTFibmdvQlFOblhpQ3ZoSDhII3o2TWtmWjZTNE5WVlRFdXRzOG81eEZ6Uk1SOGVDNlkxYm5nb0JRTm5YaUN2aEg4SCJ9"
+        }
+      ],
+      "link": "bafyreigzxkki35e3pd4r6zvowmp25fly7yx5bjqwkojwiftpe5r3xx4gnu"
+    },
+    "linkedBlock": "pGJpZNgqWCYAAYUBEiDRQJ7VCtGQWcLlmFpitGoSP35ntX7fKJeFWJ8zKz2+Z2RkYXRhgaNib3BjYWRkZHBhdGhlL21vcmVldmFsdWUY6mRwcmV22CpYJgABhQESINFAntUK0ZBZwuWYWmK0ahI/fme1ft8ol4VYnzMrPb5nZmhlYWRlcqFrY29udHJvbGxlcnOA"
+  }
+}' -H "Content-Type: application/json"
+
+{
+  "docId": "kjzl6cwe1jw14ahmwunhk9yjwawac12tb52j1uj3b9a57eohmhycec8778p3syv",
+  "state": {
+    "doctype": "tile",
+    "content": {
+      "title": "My first Document"
+    },
+    "metadata": {
+      "schema": null,
+      "controllers": [
+        "did:key:z6MkfZ6S4NVVTEuts8o5xFzRMR8eC6Y1bngoBQNnXiCvhH8H"
+      ]
+    },
+    "signature": 2,
+    "anchorStatus": "PENDING",
+    "log": [
+      {
+        "cid": "bagcqcera2faj5vik2giftqxftbngfndkci7x4z5vp3psrf4flcptgkz5xztq",
+        "type": 0
+      },
+      {
+        "cid": "bagcqcera3fkje7je4lvctkam4fvi675avtcuqgrv7dn6aoqljd5lebpl7rfq",
+        "type": 1
+      }
+    ],
+    "anchorScheduledFor": "12/30/2020, 1:15:00 PM",
+    "next": {
+      "content": {
+        "title": "My first Document",
+        "more": 234
+      },
+      "metadata": {
+        "schema": null,
+        "controllers": []
+      }
+    }
+  }
+}
+
+
+```
+
+
+
+# Pins
+
+The pins api endpoint can be used to manipulate the pin set. The pin set is all of the documents that a node maintains the state of. Any document opened by the node that is not pinned will eventually be garbage collected from the node.
+
+## Add a document to the pin set
+This method adds the document with the given [[DocID]] to the pin set. 
+
+*`⚠️ This method is not available if gateway mode is enabled.`*
+
+### Request
+`POST /api/v0/pins/:docid`
```

Comment:
huh, this does indeed match what we have in code, but is surprising to me.  Since this is a POST request not a GET, shouldn't the docid be encoded as an arg to the `pins` endpoint, rather than encoded in the URL?

CC @simonovic86